project/views.py

```diff
@@ -1,3 +1,81 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from datetime           import datetime, timedelta
+
+from django.http        import JsonResponse
+from django.views       import View
+
+from .models            import Category, Project, Gift, Community, Story, Like
+from user.models        import User
+from message.models     import Message
+
+class ProjectDetailView(View):
+    def get(self, request, project_id):
+        today = datetime.now()
+
+        if not Project.objects.filter(id=project_id).exists():
+            return JsonResponse({'message': 'PROJECT_NOT_EXIST'}, status=404)
+
+        project = Project.objects.get(id=project_id)
+
+        total_price = []
+        project_info = {
+            'category'         : project.category.name,
+            'name'             : project.name,
+            'thumbnail_url'    : project.thumbnail_url,
+            'creator'          : project.user.fullname,
+            'achieved_rate'    : project.achieved_rate,
+            'total_amount'     : project.total_amount,
+            'rest_date'        : (project.closing_date - today).days,
+            'total_supporters' : project.total_supporters,
+            'goal_amount'      : project.goal_amount,
+            'project_uri'      : project.project_uri,
+            'payment_date'     : project.closing_date + timedelta(days=1),
+            'option'           : [{
+                'option_name'   : option.name,
+                'option_price'  : option.price,
+                'quantity_sold' : option.quantity_sold,
+                'stock'         : option.stock,
+                } for option in project.gift_set.all()],
+            }
+
+        creator_info = {
+            'name'                : project.user.fullname,
+            'creator_image'       : project.user.profile_image,
+            'creator_description' : project.user.user_description
+            }
+
+        tab = {
+            'story'  : list(project.story_set.values('content')),
+            'communities' : [{
+                'user'       : community.user.fullname,
+                'comment'    : community.comment,
+                'past_date'  : (today-community.updated_at).days,
+                'created_at' : community.created_at,
+                'recomment'  : [{
+                    'user'       : community.community_set.all()[i].user.fullname,
+                    'comment'    : community.community_set.all()[i].comment,
+                    'created_at' : community.community_set.all()[i].created_at
+                    } for i in range(community.community_set.count())
```

Comment:
queryset 자체가 이미 iterable 합니다. 따라서 range를 사용하지 않고, 바로 데이터를 사용할 수 있습니다.
```suggestion
                'recomment'  : [{
                    'user'       :  community.user.fullname,
                    'comment'    : ,
                    'created_at' :  ,
                    } for community in community.community_set.all())
```

Reply:
왜........저렇게 썼을까요 ㅎㅎ 수정했습니다!